subprojects/docs/src/docs/release/notes.md

```diff
@@ -27,7 +27,23 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
 
 <!-- Do not add breaking changes or deprecations here! Add them to the upgrade guide instead. --> 
 
-## Performance improvements for incremental development
+## Performance improvements
```

Comment:
File system watching is also an improvement for incremental development. How about we keep the this header `Performance improvements for incremental development`, and call the next header `Android performance improvements` or something along those lines. WDYT?

Reply:
I'm not sure. Would you mind making your proposal a commit to this PR?

Reply:
How about now?

Reply:
Let's go for it!

Reply:
I'd keep the section name as "performance improvements" for consistency with other release notes.